app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -275,8 +277,13 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), BackHandler, Accou
     }
 
     private suspend fun deleteSelectedBookmarks(selected: Set<BookmarkNode>) {
-        selected.forEach {
-            context?.bookmarkStorage()?.deleteNode(it.guid)
+        CoroutineScope(IO).launch {
+            val tempStorage = context?.bookmarkStorage()
+            selected.forEach {
+                runBlocking {
```

Comment:
It is necessary to delete each bookmark one at a time? Maybe you could use Deferred and [awaitAll](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/await-all.html).

Reply:
@mcarare any thoughts on trying this? :)

Reply:
You mean something like this: 

```
CoroutineScope(IO).launch {
            val tempStorage = context?.bookmarkStorage()
            val deferreds = selected.map {
                async {
                    tempStorage?.deleteNode(it.guid)
                }
            }
            deferreds.awaitAll()
        }
```

Reply:
Yeah that seems reasonable to me 😄